includes/polling/ports/edgecos.inc.php

```diff
@@ -0,0 +1,33 @@
+<?php
+/**
+* edgecos.inc.php
+*
+* LibreNMS Edgecos Ports fix
+*
+* Fix the problem that EdgeCore did not follow the ifOperStatus convention
+*
+* This program is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 3 of the License, or
+* (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*
+* @package    LibreNMS
+* @link       http://librenms.org
+* @copyright  2019 Ming-Han Yang
+* @author     Ming-Han Yang <soto2080edu@g.ncu.edu.tw>
+*/
+foreach ($port_stats as & $port) {
+    if ($port['ifOperStatus'] == 'lowerLayerDown') {
```

Comment:
Why did you change this?  lowerLayerDown is a proper ifOperStatus...
If you are having issues with a rule, you can just make it check ifOperStatus = down or ifOperStatus = lowerLayerDown

Reply:
The problem is that although lowerLayerDown is defined in the MIB, but the biggest players do not follow the defines i.e Cisco always make ifOperStatus as 'down' even it should be lowerLayerDown or others in the status.

Btw, another problem is that the whole front end of the web don't have the logic that there are many type of admin up but port down.

The web front just check if port is up/up, if yes, all fine.
but if it has admin down, if yes, assign as admin down,
If not, then the problem occur, only ifOperStatus=='down' will be seem as down, all the other ifOperStatus will not be anyone of up/down/admindown, it just disappear in the frontend.

Reply:
IMHO, changing the data is not the correct way to resolve that.     Checks should probably be ifOperStatus == 'up' or ifOperStatus != 'up'.

Possible ifOperStatus status:
1: up(1)
2: down(2)
3: testing(3)
4: unknown(4)
5: dormant(5)
6: notPresent(6)
7: lowerLayerDown(7)

Possible ifAdminStatus:
1: up(1)    
2: down(2)
3: testing(3)

Reply:
I agree with you but there is no way to show these info in the frontend.